src/Pillar-Tests-Cli/ClapPillarBuildCommandTest.class.st

```diff
@@ -0,0 +1,30 @@
+Class {
+	#name : #ClapPillarBuildCommandTest,
+	#superclass : #TestCase,
+	#category : #'Pillar-Tests-Cli'
+}
+
+{ #category : #adding }
+ClapPillarBuildCommandTest >> testAddSubcommand [
+	| allIds defaultCommand |
+	defaultCommand := ClapPillarBuildCommand defaultCommand.
+	allIds := PRTarget clapCommandList collect: [ :c | c identifier ].
+	allIds do: [ :id |
+		self assert: ((defaultCommand subcommands select: [ :c | c class = ClapCommand new class and: c identifier = id asSymbol ]) size = 1).
+		 ]
+]
+
+{ #category : #tests }
+ClapPillarBuildCommandTest >> testDefaultCommandIsBuild [
+	| defaultCommand  |
+	defaultCommand := ClapPillarBuildCommand defaultCommand.
+	self assert: (defaultCommand identifier = #build)
+]
+
+{ #category : #tests }
+ClapPillarBuildCommandTest >> testSameFlagsMainAndDefault [
+	| defaultCommand mainCommand |
+	defaultCommand := ClapPillarBuildCommand defaultCommand.
+	mainCommand := PRTarget default buildClapCommand.
+	self assert: (defaultCommand flags) asString = (mainCommand flags) asString
```

Comment:
1. Can't we compare flags directly? Why converting them to string?
2. you should use `#assert:equals:` instead of `#assert: =`. I'm sure the code quality rules gave you a warning about this :)

Reply:
Technically the flags could be shared and compared by identity, but I'm not sure it's worth the hassle…

Reply:
Equality is enough to me :).
Ensuring we use identical flags will complexify the construction for not much gain.
We have to be careful with meanings for example, not only descriptions and ids.

Reply:
1. I tried to compare them directly as objects, but it failed. Though when I inspect both OrderedCollections, they contain same flags... This is why I introduced the "asString"
2. Yes ^^ I replaced by equals: :)